File: HW_2/plane.cpp
#include "plane.h"

int RandInt(int min, int max);

// Read plane information from file.
void plane::Read(FILE *file) {
    transport::Read(file);
    fscanf(file, "%d", &this->maxDistance);
    fscanf(file, "%d", &this->capacity);
}

// Write plane information to file.
void plane::Write(FILE *file) {
    fprintf(file, "%s", "plane ");
    transport::Write(file);
    fprintf(file, "%c%d%c%d", ' ', this->maxDistance, ' ', this->capacity);
}

// Calculate time to cover the distance.
double plane::MeasureTime() {
    if (this->maxDistance < this->distance) {
        return 0;
    }
    return this->distance / this->velocity;
}

// Generate random plane.
void plane::Generate() {
    transport::Generate();
    this->maxDistance = RandInt(1, 1000);
    this->capacity = RandInt(0, 1000);
}

File: HW_3/Plane.py
from Transport import Transport
import random


class Plane(Transport):
    def __init__(self, params=None):
        super().__init__(params)
        if params is None:
            self.max_distance = 0
            self.capacity = 0
        else:
            self.max_distance = int(params[2])
            self.capacity = int(params[3])

    def generate(self):
        super().generate()
        self.max_distance = random.randint(950, 2000)
        self.capacity = random.randint(1, 15)
        return self

    def measure_time(self) -> float:
        if self.distance > self.max_distance:
            return 1e100 + self.distance
        return super().measure_time()

    def __str__(self):
        return f"Plane {super().__str__()} {self.max_distance} {self.capacity}"
